aaad/authorizers.py

```diff
@@ -127,9 +131,73 @@ def authorize(self, user, act_as, resource, logging, info, data, content_type, a
 
             return True
 
+        def get_user_list(self, type=None):
+            if not type:
+                return self.data.keys()
+            users = []
+            for key, data in self.data.items():
+                if data.get('type', 'user') == type:
+                    users.append(key)
+            return users
+
+        def get_canactas_list(self, user):
+            actas_users = []
+            for key in self._get_act_as_list(user):
+                if (key != user and
+                  self.data.get(key).get('type', 'user') != 'internal'):
+                    actas_users.append(key)
+            return actas_users
+
+        def is_user_valid(self, user):
+            return user in self.data.keys()
+
+        def filter_response(self, resource, data, actas):
+            framework = FrameworkUtils().getFramework(resource)
+            allowed_namespaces = self._get_allowed_namespace_patterns(actas, self.data)
+            if not allowed_namespaces:    #Empty list
+                return ""
+            return framework.filterResponseBody(data, allowed_namespaces, resource)
+
+        def _parse_permissions_file(self, filename):
+            with open(filename, 'r') as data_file:
+                return yaml.load(data_file)
+            return ''
+
+        def _get_merged_data(self, user, allowed_users, allowed_actions, data, action):
+            if user in allowed_users:
+                return
+            allowed_users.append(user)
+            if action != '' and 'action' in data[user]:
+                if data[user]['action'] != None and action in data[user]['action']:
+                    for item in data[user]['action'][action]:
+                        temp_item = {}
+                        if type(item) == str:
+                            temp_item = {}
+                            temp_item[item] = {}
+                        else:
+                            if type(item) == dict:
+                                temp_item = item
+                        if not temp_item in allowed_actions:
+                            allowed_actions.append(temp_item)
+            if 'can_act_as' not in data[user]:
+                return
+            for u in data[user]['can_act_as']:
+                self._get_merged_data(u, allowed_users, allowed_actions, data, action)
+
+        def _get_allowed_namespace_patterns(self, act_as, permissions):
+            allowed_users_list = []
+            self._get_merged_data(act_as, allowed_users_list, [], permissions, '')
+            allowed_namespace_patterns = []
+            for user in allowed_users_list:
+                if 'allowed_names' in permissions[user]:
+                    for pattern in permissions[user]['allowed_names']:
+                        if not pattern in allowed_namespace_patterns:
+                            allowed_namespace_patterns.append(pattern)
+            return allowed_namespace_patterns
+
```

Comment:
Nice small methods. Smaller methods are are easier to unit test than bigger ones.